python/grass/pygrass/modules/grid/grid.py

```diff
@@ -154,10 +154,6 @@ def get_mapset(gisrc_src, gisrc_dst):
         copy_special_mapset_files(path_src, path_dst)
     src = Mapset(msrc, lsrc, gsrc)
     dst = Mapset(mdst, ldst, gdst)
-    visible = [m for m in src.visible]
-    if src.name not in visible:
-        visible.append(src.name)
-    dst.visible.extend(visible)
```

Comment:
I believe we should keep this. If I interpret this correctly, this is trying to replicate search path of the current mapset in the temporary mapset. So if user doesn't specify input with full name@mapset, this would be needed. At least it doesn't harm, since it's a temp mapset anyway.

Reply:
I originally made it so that the user didn't have to specify the exact MAPSET name (full name@MAPSET) when entering the input map name, but then the search path of the current MAPSET needs to be reseted after the `run()` method completes.

https://github.com/OSGeo/grass/pull/2567#pullrequestreview-1107115943

Reply:
> I believe we should keep this. If I interpret this correctly, this is trying to replicate search path of the current mapset in the temporary mapset. So if user doesn't specify input with full name@mapset, this would be needed. At least it doesn't harm, since it's a temp mapset anyway.

Example (if these lines are preserved):

1. Launch GG with with **nc_spm_08_grass7** LOCATION and **landsat** MAPSET
2. `g.mapsets operation=set mapset=landsat`
```
GRASS nc_spm_08_grass7/landsat:~ > g.mapsets -p
Accessible mapsets:
landsat
```
3. Launch Python script mentioned [here](https://github.com/OSGeo/grass/pull/2567#issue-1359409831) 
4. Check current MAPSET search path again
```
GRASS nc_spm_08_grass7/landsat:~ > g.mapsets -p
Accessible mapsets:
landsat PERMANENT
```

Reply:
Tested with #2584 and these code lines are preserved (which you mentioned above):

4. Check current MAPSET search path again

```
GRASS nc_spm_08_grass7/landsat:~ > g.mapsets -p
Accessible mapsets:
landsat PERMANENT climate_2000_2012
```